app/javascript/audio-recorder.js

```diff
@@ -1,63 +1,74 @@
 import { DirectUpload } from "activestorage"
 
-function responseFileName (codec) {
-  const fileTypes = {
-    'audio/wav': 'wav',
-    'audio/webm;codecs=opus': 'webm',
-    'audio/ogg;codecs=opus': 'ogg'
-  };
+(() => {
+
+const fileTypes = {
+  'audio/wav': 'wav',
+  'audio/ogg;codecs=opus': 'ogg',
+};
 
+function responseFileName (codec) {
   return `response-${Date.now()}.${fileTypes[codec]}`;
 }
 
 document.addEventListener("DOMContentLoaded", function(){
-  const input = document.querySelector('#story_audio')
-  const recordButton = document.getElementById("recordButton")
-  const stopButton = document.getElementById("stopButton")
-  let recorder
-
-  const uploadFile = (file) => {
-    const url = input.dataset.directUploadUrl
-    const upload = new DirectUpload(file, url)
-
-    upload.create((error, blob) => {
-      if (error) {
-        const errorAlert = document.createElement('div')
-        errorAlert.textContent = 'Sorry, there was an error submitting your audio.'
-        document.querySelector('.content > .container > .content-column').appendChild(errorAlert)
-      } else {
-        // Add an appropriately-named hidden input to the form with a
-        //  value of blob.signed_id so that the blob ids will be
-        //  transmitted in the normal upload flow
-        const hiddenField = document.createElement('input')
-        hiddenField.setAttribute("type", "hidden")
-        hiddenField.setAttribute("value", blob.signed_id)
-        hiddenField.name = input.name
-        document.querySelector('form').appendChild(hiddenField)
-      }
-    })
+  const audioContainer = document.querySelector('#audio-recording')
+  const audioUnsupported = Object.keys(fileTypes).every(type => !MediaRecorder.isTypeSupported(type));
+
+  if (audioUnsupported || !audioContainer) {
+    audioContainer && audioContainer.parentNode.removeChild(audioContainer);
   }
+  else {
+    const input = audioContainer.querySelector('#story_audio')
+    const recordButton = audioContainer.getElementById("recordButton");
```

Comment:
`getElementById` was not available to the audioContainer object because it does not have that method. It's only available on `document`.